### DISCUSSION TITLE
make CocDiagnostics sticky

### DISCUSSION THREAD

**Discussion by laoshaw** [neoclide/coc.nvim#4844] [General]
I saw the diagnostic messages shown at the bottom window then it disappears in a few seconds, I can use :CocDiagnostics to bring that back, is there a way to make the warning/error terminal window sticky until all errors/warnings are fixed?

**Comment by fannheyward** [comment]
Try `:CocList diagnostics`, this lists diagnostics and will refresh after be fixed.

**Comment by laoshaw** [comment]
thanks, this is similar to :CocDiagnostics, in both cases, if there are errors the message window is not auto open, I need type :CocDiagnostics or :CocList diagnostics to bring that up. is there a way like ALE's ale_open_list that will keep the quickfix window open automatically, so I know my code has issues and will fix them before I save the final version? seems like a really handy default for daily coding.

**Reply by fannheyward** [reply]
> keep the quickfix window open automatically

Maybe `autocmd`?

**Reply by laoshaw** [reply]
it's actually location window instead of quickfix quick window, i wrote a autocmd to run lclose and it works.

each time I need save the file to refresh a new error/warning checking-and-update, is there a way for coc.nvim to check any syntax errors while I'm editing the code without saving it, or is this a clangd thing?

ALE for example does that, including auto-open the warning/error message window, but I'm trying to go coc.nvim all the way instead of mixing two.
